dask_expr/collection.py

```diff
@@ -707,6 +709,21 @@ def select_dtypes(self, include=None, exclude=None):
     def eval(self, expr, **kwargs):
         return new_collection(Eval(self.expr, _expr=expr, expr_kwargs=kwargs))
 
+    def set_index(self, other, drop=True, sorted=False, divisions=None):
+        if isinstance(other, Series):
+            if other._name == self.index._name:
+                return self
+        else:
+            if other == self.index.name:
+                return self
+
+        if divisions is not None:
+            check_divisions(divisions)
+        other = other.expr if isinstance(other, Series) else other
```

Comment:
Might a user do `df.set_index(some_dataframe)`?

Reply:
Yep potentially, we should disallow this though

Reply:
Added a check